nselib/httpcookies.lua

```diff
@@ -0,0 +1,476 @@
+
+    if ( o:parse(self.cookies) ) then
+      return o
+    end
+
+  end,
+
+
+  --- Parses the cookie and and splits it into its attributes if its a string.
+  -- @param cookies A cookie table 
+  -- @return status true on success, false on failure
+  parse = function(cookies)
+    if cookies == nil then
+      return true
+    end
+    if (type(cookies) == 'table') then
+      --Name and Value must be present in the cookie, rest
+      --attributes are optional.
+      --So, if we have a table of cookies, we take the first table 
+      --and then loop for all the attributes in that table.
+      for _,cookie in ipairs(cookies) do
+        for cookie_attribute, cookie_attribute_value in pairs(cookie) do
+          cookie_attribute = cookie_attribute:lower()
+          if(cookie_attribute == 'name') then 
+          if(type(cookie_attribute_value) ~= 'string') then
+            stdnse.debug1("Cookie name is not a string")
+            return false --Name has to be of type string
+        end
+          elseif(cookie_attribute == 'value') then
+          if(type(cookie_attribute_value) ~= 'string') then
+            stdnse.debug1("Cookie value is not a string")
+            return false
+          end
+          elseif(cookie_attribute == 'path') then
+            if(type(cookie_attribute_value) ~= 'string') then
+              stdnse.debug1("Cookie path is not a string")  
+            return false
+          end
+        elseif(cookie_attribute == 'expires') then
+          if(type(cookie_attribute_value) ~= 'string') then
+            stdnse.debug1("Cookie expires attribute is not a string") 
+            return false
+          end
+        elseif(cookie_attribute == 'max-age') then
+          if(type(cookie_attribute_value) ~= 'string') then
+            stdnse.debug1("Cookie max-age attribute is not a string") 
+            return false
+          end
+        elseif(cookie_attribute == 'domain') then
+          if(type(cookie_attribute_value) ~= 'string') then
+            stdnse.debug1("Cookie domain attribute is not a string")  
+            return false
+          end
+        elseif(cookie_attribute == 'secure') then 
+          if(type(cookie_attribute_value) ~= 'boolean') then
+            stdnse.debug1("Cookie secure attribute is not a boolean")
+            return false
+          end
+        elseif(cookie_attribute == 'httponly') then
+          if(type(cookie_attribute_value) ~= 'boolean') then
+            stdnse.debug1("Cookie httponly attribute is not a boolean")
+            return false
+          end
+        else 
+          stdnse.debug1("Cookie attribute is not recognised")
+          return false
+        end
+      end
+    end 
+    return true
+  elseif (type(cookies) == 'string') then 
+      --We can parse it using the http parse cookie function
+      cookies, status = http.parse_set_cookie(cookies)
+      --Does it make sense to have multiple cookies parsed when we are using a string argument(?)
```

Comment:
Let's set a limit to one cookie if we are using a string. Make sure this behavior is documented though.